### PR TITLE
Base qtouch

### PR DESCRIPTION
**Pull request by NickeZ**

### CODE REVIEW

**Comment by NickeZ** [comment]
To test this you can build and run the `fw_bitboxbase_touch_demo.elf`. The jlink script to flash is in `build/test/device-test`.

File: src/hardfault.c [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #include "hardfault.h"
 #include "util.h"
+#include <platform_config.h>
```
Comment (conte91) [inline src/hardfault.c]: This should go in a separate commit I guess?

File: src/touch/gestures.c [hunk 2]
```diff
@@ -44,7 +46,7 @@ extern volatile uint8_t measurement_done_touch;
 
 /********************************** STATE **********************************/
 
-static const uint16_t LONG_TOUCH = 500;
+static const uint16_t LONG_TOUCH = 500 * 2;
```
Comment (conte91) [inline src/touch/gestures.c]: ...1000? Also this will affect the BB02 behaviour!

File: src/touch/gestures.h [hunk 3]
```diff
@@ -39,6 +39,11 @@ typedef struct {
     int32_t velocity;
 } gestures_slider_data_t;
 
+enum button_id_t {
```
Comment (conte91) [inline src/touch/gestures.h]: `bitboxbase_button_id_t`? Or guard it with `#ifdef`s so that we're sure it doesn't affect BB02 stuff.

File: src/touch/gestures.h [hunk 4]
```diff
@@ -39,6 +39,11 @@ typedef struct {
...
+    BUTTON_LEFT,
```
Comment (conte91) [inline src/touch/gestures.h]: `BITBOXBASE_BUTTON_{LEFT,RIGHT}`

File: src/hardfault.c (hunk 1 above)
Comment (NickeZ) [reply src/hardfault.c]: It is ;)

File: src/touch/gestures.c (hunk 2 above)
Comment (NickeZ) [reply src/touch/gestures.c]: ack

File: src/touch/gestures.h (hunk 3 above)
Comment (NickeZ) [reply src/touch/gestures.h]: ack

File: src/touch/gestures.h (hunk 4 above)
Comment (NickeZ) [reply src/touch/gestures.h]: ack
